### DISCUSSION TITLE
no custom domain name option available in gh pages

### DISCUSSION THREAD

**Discussion by fireflysemantics** [community/community#22419] [Other Feature Feedback, Questions, & Ideas]
<p>I just created this repository:</p>
<p><a href="https://github.com/fireflysemantics/test-deploy.fireflysemantics.com" rel="nofollow noopener">https://github.com/fireflysemantics/test-deploy.fireflysemantics.com</a></p>
<p>I then setup A records for it via my domain provider.  After that I pushed  test page to gh-pages.  When I go to settings &gt; gh-pages it just says:</p>
<p> Your GitHub Pages site is currently being built from the gh-pages branch.</p>
<p>And there is no custom domain option.  I wish to serve the url <code>test-deploy.fireflysemantics.com</code> and on the other repositories that I’m serving from a sub domain this is all possible.  Thoughts?</p>

**Comment by ghost** [comment, marked as answer]
<p>Hey <a class="mention" href="/fireflysemantics">@fireflysemantics</a>,</p>
<p>Helps arrived! The  <strong>gh-pages</strong> that GitHub is refering to is the branch of the repository that is deployed, not the URL that it’s deployed to. Under that section, there should be a section that is titled <strong>Custom domain</strong>. Here, you can enter the custom domain that you wish for the site to be deployed to. As long as you’ve configured the DNS correctly (which it looks like you have), then it should push to that domain.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/3/398cfa883495f7826760c8300b9eba5c909a5d30.png" data-download-href="/uploads/short-url/8d7gbTS7e3rQteLumdyE0NWGzmw.png?dl=1" title="GitHub Pages - Custom domains.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/22669894/181070831-14349c78-3ed1-4751-977f-534f92cfbae7.png" alt="GitHub Pages - Custom domains.png" data-base62-sha1="8d7gbTS7e3rQteLumdyE0NWGzmw" width="550" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/398cfa883495f7826760c8300b9eba5c909a5d30_2_550x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/3/398cfa883495f7826760c8300b9eba5c909a5d30_2_825x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/3/398cfa883495f7826760c8300b9eba5c909a5d30_2_1100x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/398cfa883495f7826760c8300b9eba5c909a5d30_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">GitHub Pages - Custom domains.png</span><span class="informations">1518×1380 140 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>There’s some more info about this <a href="https://help.github.com/en/articles/adding-or-removing-a-custom-domain-for-your-github-pages-site" rel="nofollow noopener">over here</a>.</p>

**Comment by carlogilmar** [comment]
<p>I know that you have to create a file named CNAME with the custom domain, and in the DNS you have to redirect the domain to GitHub :slight_smile:</p>

**Comment by JustinGOSSES** [comment]
<p>I have the same problem. There is no “custom domain” form field on GitHub Enterprise Server version 3.0.12 as described in the documentation for GitHub.</p>
<p>It is not clear in what GitHub products this form field will or will not exist and whether it is just not available in some products or configurable to be turned off as an option. I have admin right and can’t find any configuration to turn off or on the CNAME form field in GitHub Enterprise 3.0.12.</p>

**Reply by ImanMahmoudinasab** [reply]
@JustinGOSSES Did you figure out how to set a custom domain on Github Enterprise Server?

**Comment by tcbyrd** [comment]
<p><a class="mention" href="/justingosses">@JustinGOSSES</a> Custom domains are not possible in the on-prem version of GitHub Enterprise server. All pages sites are accessible at either <code>pages.&lt;hostname&gt;</code> or <code>&lt;hostname&gt;/pages</code>, depending on whether you have subdomain isolation enabled. For various architectural reasons, it’s not feasible to support custom domains for Pages sites in GHES.</p>
<p>In case it helps you or anyone else that come across this post in the future, for this reason and a few others, we’ve added a few features on <a href="http://github.com">github.com</a> to help make Pages work better for Enterprise use-cases:</p>
<aside class="onebox allowlistedgeneric" data-onebox-src="https://docs.github.com/en/enterprise-cloud@latest/pages/getting-started-with-github-pages/changing-the-visibility-of-your-github-pages-site">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">

      <a href="https://docs.github.com/en/enterprise-cloud@latest/pages/getting-started-with-github-pages/changing-the-visibility-of-your-github-pages-site" target="_blank" rel="noopener">GitHub Docs</a>
  </header>

  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/13207348/181070833-073547e7-ca7b-44e5-a05b-2e42a5f4be70.png" class="thumbnail onebox-avatar" width="500" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_500x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_750x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_1000x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_10x10.png">

<h3><a href="https://docs.github.com/en/enterprise-cloud@latest/pages/getting-started-with-github-pages/changing-the-visibility-of-your-github-pages-site" target="_blank" rel="noopener">Changing the visibility of your GitHub Pages site - GitHub Docs</a></h3>



  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>
<aside class="onebox allowlistedgeneric" data-onebox-src="https://docs.github.com/en/enterprise-cloud@latest/organizations/managing-organization-settings/managing-the-publication-of-github-pages-sites-for-your-organization">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">

      <a href="https://docs.github.com/en/enterprise-cloud@latest/organizations/managing-organization-settings/managing-the-publication-of-github-pages-sites-for-your-organization" target="_blank" rel="noopener">GitHub Docs</a>
  </header>

  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/13207348/181070834-6f6f52e5-a855-422b-890b-74136bc009ed.png" class="thumbnail onebox-avatar" width="500" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_500x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_750x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_1000x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_10x10.png">

<h3><a href="https://docs.github.com/en/enterprise-cloud@latest/organizations/managing-organization-settings/managing-the-publication-of-github-pages-sites-for-your-organization" target="_blank" rel="noopener">Managing the publication of GitHub Pages sites for your organization - GitHub...</a></h3>



  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>
<aside class="onebox allowlistedgeneric" data-onebox-src="https://docs.github.com/en/enterprise-cloud@latest/pages/configuring-a-custom-domain-for-your-github-pages-site/verifying-your-custom-domain-for-github-pages">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">

      <a href="https://docs.github.com/en/enterprise-cloud@latest/pages/configuring-a-custom-domain-for-your-github-pages-site/verifying-your-custom-domain-for-github-pages" target="_blank" rel="noopener">GitHub Docs</a>
  </header>

  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/13207348/181070835-36be5be3-9f8b-43be-bc08-d7cb7e7f3d56.png" class="thumbnail onebox-avatar" width="500" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_500x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_750x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_1000x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_10x10.png">

<h3><a href="https://docs.github.com/en/enterprise-cloud@latest/pages/configuring-a-custom-domain-for-your-github-pages-site/verifying-your-custom-domain-for-github-pages" target="_blank" rel="noopener">Verifying your custom domain for GitHub Pages - GitHub Docs</a></h3>



  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>So if you need a Pages site with a custom domain that is only accessible to people in your Enterprise, you can do this now on <a href="http://github.com">github.com</a>.</p>
